sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -576,57 +563,71 @@ private void OverrideBindingName(INamedTypeSymbol attributeClass, ref string arg
                 }
             }
 
+            private bool IsCardinalitySupported(AttributeData attribute)
+            {
+                var attrClass = attribute.AttributeClass;
+                var isBatchedProp = attrClass!.GetMembers().Where(m => string.Equals(m.Name, Constants.FunctionMetadataBindingProps.IsBatchedKey, StringComparison.OrdinalIgnoreCase)).SingleOrDefault();
+
+                if (isBatchedProp != null)
+                {
+                    return true;
+                }
+
+                return false;
+            }
+
             /// <summary>
-            /// This method verifies that an EventHubsTrigger matches our expectations on cardinality (isBatched property). If isBatched is set to true, the parameter with the
+            /// This method verifies that a binding that has Cardinality (isBatched property) is valid. If isBatched is set to true, the parameter with the
             /// attribute must be an enumerable type.
             /// </summary>
-            private bool IsEventHubsTriggerValid(IParameterSymbol parameterSymbol, TypeSyntax? parameterTypeSyntax, SemanticModel model, AttributeData attribute, out DataType dataType, out Cardinality cardinality)
+            private bool IsCardinalityValid(IParameterSymbol parameterSymbol, TypeSyntax? parameterTypeSyntax, SemanticModel model, AttributeData attribute, out DataType dataType)
```

Comment:
Similar to the legacy generator [here](https://github.com/Azure/azure-functions-dotnet-worker/blob/584cab006b82e0cbc2b95a8b66e2546d3a8ddfe4/sdk/Sdk/FunctionMetadataGenerator.cs#LL546C36-L546C36), for iterable collections, we also have to find the core data type while verifying that the type is an iterable collection.

Reply:
so the value set in 218 will be overwritten? Ex: If the type was List<MyPoco> it will be first List<T> and then this out param set it to MyPoco? Should we use 2 separate variables for those?